azure/compute/src/main/java/org/jclouds/azurecompute/compute/strategy/AzureComputeServiceAdapter.java

```diff
@@ -0,0 +1,326 @@
+        configset.setAdminUserName(adminUserName);
+        configset.setEnableAutomaticUpdates(false);
+        configlist.add(configset);
+        URI mediaLinkUriValue = null;
+        try {
+            mediaLinkUriValue = new URI("http://" + templateOptions.getStorageAccount() + ".blob.core.windows.net/vhd-store/" + vmName + ".vhd");
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+        }
+        String operatingSystemName = "Windows";
+
+        OSVirtualHardDisk oSVirtualHardDisk = new OSVirtualHardDisk();
+        //required
+        oSVirtualHardDisk.setName(vmName);
+        oSVirtualHardDisk.setHostCaching(VirtualHardDiskHostCaching.ReadWrite);
+        oSVirtualHardDisk.setOperatingSystem(operatingSystemName);
+        //required
+        oSVirtualHardDisk.setMediaLink(mediaLinkUriValue);
+        //required
+        oSVirtualHardDisk.setSourceImageName(template.getImage().getId());
+
+        // check if deployment already exists; if not, create deployment, otherwise add role
+        if (hostedService == null || hostedService.getDeployments().isEmpty()) {
+            VirtualMachineCreateDeploymentParameters parameters = new VirtualMachineCreateDeploymentParameters();
+            parameters.setName(group);
+            parameters.setDeploymentSlot(DeploymentSlot.Production);
+            parameters.setLabel(vmName);
+            ArrayList<Role> roles = new ArrayList<Role>();
+//
+            Role role = new Role();
+//
+
+
+//        oSVirtualHardDisk.setSourceImageName(sourceImageName);
+            //required
+            role.setRoleName(vmName);
+            //required
+            role.setRoleType(VirtualMachineRoleType.PersistentVMRole.toString());
+            role.setRoleSize(template.getHardware().getId());
+//        role.setRoleSize(VirtualMachineRoleSize.Medium);
+            role.setProvisionGuestAgent(true);
+            role.setConfigurationSets(configlist);
+            role.setOSVirtualHardDisk(oSVirtualHardDisk);
+
+            roles.add(role);
+            parameters.setRoles(roles);
+//        client.getServerApi().addDeployment(group, deploymentToParameters.apply(deployment));
+            client.getVirtualMachineApi().addDeployment(group, parameters);
+        } else {
+            VirtualMachineCreateParameters parameters = new VirtualMachineCreateParameters();
+            parameters.setRoleName(vmName);
+            parameters.setRoleSize(template.getHardware().getId());
+            parameters.setOSVirtualHardDisk(oSVirtualHardDisk);
+            parameters.setProvisionGuestAgent(true);
+            parameters.setConfigurationSets(configlist);
+
+            client.getVirtualMachineApi().addRole(group, hostedService.getDeployments().get(0).getName(), parameters);
+        }
+
+        VirtualMachine virtualMachine = getNode(name);
+        virtualMachine.setServiceName(group);
+        return new NodeAndInitialCredentials<VirtualMachine>(virtualMachine, name, null);
```

Comment:
The node credentials must be returned, as they will be used by the compute service to bootstrap the node, run scripts, etc.